File: src/ZoneCodeGenerator/Domain/Evaluation/OperandStatic.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ZoneCodeGenerator.Domain.Evaluation
{
    class OperandStatic : IEvaluation
    {
        public int Value { get; }
        public EnumMember EnumMemberValue { get; }

        public bool IsStatic => true;
        public int EvaluateNumeric()
        {
            return Value;
        }

        public OperandStatic(int value)
        {
            Value = value;
            EnumMemberValue = null;
        }

        public OperandStatic(EnumMember enumMember)
        {
            Value = (int) enumMember.Value;
            EnumMemberValue = enumMember;
        }

        public override string ToString()
        {
            return Value.ToString();
        }
    }
}

File: src/ZoneCodeGeneratorNew/Domain/Evaluation/OperandStatic.cpp
#include "OperandStatic.h"

OperandStatic::OperandStatic(const int value)
    : m_value(value),
      m_enum_member(nullptr)
{
}

OperandStatic::OperandStatic(const int value, EnumMember* enumMember)
    : m_value(value),
      m_enum_member(enumMember)
{
}

EvaluationType OperandStatic::GetType() const
{
    return EvaluationType::OPERAND_STATIC;
}

bool OperandStatic::IsStatic() const
{
    return true;
}

int OperandStatic::EvaluateNumeric() const
{
    return m_value;
}
